packages/apollo-engine-reporting/src/extension.ts

```diff
@@ -70,6 +74,15 @@ export class EngineReportingExtension<TContext = any>
       'metrics' | 'queryHash'
     >;
   }): EndHandler {
+    // If we don't want to report a query return early with an empty end handler
+    // We set `captureTraces` because that information is used to determine whether
+    // to send headers down to federated services from the gateway
+    if (!this.shouldReportQuery(o.requestContext)) {
```

Comment:
Per my comment above about `shouldReportQuery` receiving `GraphQLRequestContextDidResolveOperation` (and, to be clear, this isn't possibly with the `graphql-extensions` API that this is currently based on), I think we should invoke this within `didResolveOperation`.

Additionally, I think it should be `await`ed so that `async` behaviors can take place within?  This would be useful when, e.g., checking with an external service/shared bus to determine the trace-ability at a much earlier stage (even before Apollo Server).